atom/browser/api/atom_api_window.cc

```diff
@@ -584,8 +584,17 @@ void Window::SetFocusable(bool focusable) {
   return window_->SetFocusable(focusable);
 }
 
-void Window::SetProgressBar(double progress) {
-  window_->SetProgressBar(progress);
+void Window::SetProgressBar(double progress, mate::Arguments* args) {
+  std::string mode;
+
+  if (args && args->Length() == 2) {
```

Comment:
`args` is guaranteed to be valid, this can be simplified as:

```c++
mate::Dictionary options;
std::string mode;
args->GetNext(&options) && options.Get("mode", &mode);
```